subsys/bluetooth/shell/bt.c

```diff
@@ -1945,7 +1944,7 @@ static int cmd_adv_data(const struct shell *sh, size_t argc, char *argv[])
 			}
 
 			data = sd;
-			data_len = &sd_len;
+			data_len = sd_len;
```

Comment:
Hmm, looking over the code again, it looks like `data` and `data_len` is intended to let the user control if the data set is set for the scan response or advertising data. 

Line 1972 assumes that `data_len` is for `ad` only. 

So while this PR (so far) fixes an issue with `data` being uninitialized, it modifies the behavior of the command, so that it is actually impossible to set scan response data, as `sd_len` will always be `0` after this change. 

I suggest to initialize `data` to NULL instead (with checks before it is being dereferenced), and modify line 1972 to use `data` instead of `ad`, or something other to ensure that `data_len` isn't assumed to be the length of just the `ad`

Reply:
This is indeed a problem of scan response.
I‘ve fixed it to use`data` instead of `ad` in `ad_int`  and assigne the correct value of 'sd_len' .